pkg/operation/botanist/component/kubeapiserver/configmaps.go

```diff
@@ -79,6 +85,89 @@ func admissionPluginsConfigFilename(name string) string {
 	return strings.ToLower(name) + ".yaml"
 }
 
+func computeRelevantAdmissionPluginRawConfig(plugin AdmissionPluginConfig) ([]byte, error) {
+	var (
+		nothingToMutate    = (plugin.Config == nil || plugin.Config.Raw == nil) && len(plugin.Kubeconfig) == 0
+		mustDefaultConfig  = (plugin.Config == nil || plugin.Config.Raw == nil) && len(plugin.Kubeconfig) > 0
+		kubeconfigFilePath = volumeMountPathAdmissionKubeconfigSecrets + "/" + admissionPluginsKubeconfigFilename(plugin.Name)
+	)
+
+	if len(plugin.Kubeconfig) == 0 {
+		// This makes sure that the path to the kubeconfig is overwritten if specified in case no kubeconfig was
+		// provided. It prevents that users can access arbitrary files in the kube-apiserver pods and disguise them as
+		// kubeconfigs for their admission plugin configs.
```

Comment:
Just one question, how can users disguise arbitrary files as kubeconfigs for the plugins? We don't add it to the admissionkubeconfig secret if `plugin.Kubeconfig` is empty.

Reply:
Users could provide admission config which already contains the `kubeConfig: /some/path/to/some/file` key. Thus, without protection, they might be able to access arbitrary files in the `kube-apiserver` pod they should not have access to.

Reply:
Ah, got it!

Reply:
So, previously this was allowed, right? To provide a random file in the `kubeConfig` field config.

Reply:
I'm afraid yes